feat: CI Pipeline to build and test CPP SDK core module

AdityaKasar opened the pull request.

kschrief commented on .github/workflows/cxx-build.yml:
```diff
@@ -0,0 +1,60 @@
+name: CXX build
+on:
+  workflow_dispatch:
+  push:
```
We would want these running on PRs, not pushes.

kschrief commented on .github/workflows/utils.sh:
```diff
@@ -152,15 +154,214 @@ function unzipArtifact(){
+  cd firebolt-core-native-sdk-${FIREBOLT_VERSION}/
+  chmod +x ./build.sh
+  sed -i -e 's/prefix=/prefix /g' build.sh
+  cat ./build.sh
```
Why are we "cat"ing this file and others below?

kschrief commented on .github/workflows/utils.sh:
```diff
@@ -152,15 +154,214 @@ function unzipArtifact(){
+  cmake --build build/Thunder --target install && echo "Thunder Build succeeded" || exit 9999
+}
+
+function buildCoreCPPSDK() {
```
Can we combine the 3 build functions into one function that takes in an SDK parameter? We'll have other SDKs to worry about down the road and it'll be easier to add it as a parameter.

kschrief replied:
I also don't see these functions being used anywhere.

kschrief commented on .github/workflows/cxx-build.yml:
```diff
@@ -0,0 +1,60 @@
+          ls
+          npm run cpp
+
+      - name: Build SDK
```
It looks like we're only building core here? Can we also build the other SDKs?

AdityaKasar commented:
Requesting a re-review 

- Addressed previous comments 
- Implemented caching mechanism for faster installation of npm dependencies 
- Pre-compiling Thunder and openrpc artifacts and used artifacts uploading feature 
- Building 3 SDKs parallelly 

This is the final pipeline for reference: https://github.com/rdkcentral/firebolt-apis/actions/runs/11291931107

AdityaKasar commented:
latest pipeline run: https://github.com/rdkcentral/firebolt-apis/actions/runs/11354300763/job/31581481906

kschrief commented:
:tada: This PR is included in version 1.5.0-next.2 :tada:

The release is available on [GitHub release](https://github.com/rdkcentral/firebolt-apis/releases/tag/v1.5.0-next.2)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

kschrief commented:
:tada: This PR is included in version 1.5.0 :tada:

The release is available on [GitHub release](https://github.com/rdkcentral/firebolt-apis/releases/tag/v1.5.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
